pkg/ccl/backupccl/restore_job.go

```diff
@@ -280,8 +282,41 @@ func restore(
 	// which are grouped by keyrange.
 	highWaterMark := job.Progress().Details.(*jobspb.Progress_Restore).Restore.HighWater
 
-	importSpans := makeSimpleImportSpans(dataToRestore.getSpans(), backupManifests,
-		backupLocalityMap, introducedSpanFrontier, highWaterMark, targetRestoreSpanSize.Get(execCtx.ExecCfg().SV()))
+	var importSpans []execinfrapb.RestoreSpanEntry
+	var checkpointingSpanFrontier *spanUtils.Frontier
+	if execCtx.ExecCfg().Settings.Version.IsActive(restoreCtx, clusterversion.V23_1Start) {
+		spanFrontierSlice := job.Progress().Details.(*jobspb.Progress_Restore).Restore.Frontier
+		var frontierCreationSlice []roachpb.Span
+		for _, frontierSpanEntry := range spanFrontierSlice {
+			frontierCreationSlice = append(frontierCreationSlice, frontierSpanEntry.Entry)
+		}
+		checkpointingSpanFrontier, err = spanUtils.MakeFrontier(frontierCreationSlice...)
+		for _, frontierSpanEntry := range spanFrontierSlice {
+			_, err = checkpointingSpanFrontier.Forward(frontierSpanEntry.Entry, frontierSpanEntry.Timestamp)
+			if err != nil {
+				return emptyRowCount, err
+			}
+		}
+		importSpans = makeSimpleImportSpans(dataToRestore.getSpans(), backupManifests,
```

Comment:
could you explain why you need this spanFrontierLength? why not just pass in the span frontier?

Reply:
Believe it was to handle the empty case of using the new logic but we have no spans in the span frontier, however I believe we can omit this case since explicitly checking the span frontier == nil should handle this